src/fast/u192.rs

```diff
@@ -0,0 +1,57 @@
+use std::mem::transmute;
+use std::ops::{Add, Mul};
+
+// This was started by the output of the uint crate,
+// then heavily reduced to only the parts we need
+// (which was a significant optimization, especially
+// in the mul which carried out the full multiplication
+// and discarded the top-half)
+//
+/// Little-endian large integer type
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub struct U192(pub [u64; 3]);
+
+impl Mul for U192 {
+    type Output = Self;
+    fn mul(self, other: Self) -> Self {
+        // The generated implementation of this method was 360 lines long!
+        let me = &self.0;
+        let you = &other.0;
+
+        let mult = |m: usize, y: usize| unsafe {
```

Comment:
`usize` here is a bit strange - shouldn't this make it clear that we are multiplying two `u64` to get a `u128`?

Reply:
The `usize` is because `mult(0, 1)` expands to `transmute::<_, (u64, u64)>(me[0] as u128 * you[1] as u128)` (arguments are indexes to `me` and `you`... it just makes it more terse. I could make this `mult(me[0], you[1])` if you think that is better / more readable.